presto-main/src/main/java/com/facebook/presto/metadata/FunctionAndTypeManager.java

```diff
@@ -256,14 +276,14 @@ public CacheStatsMBean getFunctionResolutionCacheStats()
     public void loadFunctionNamespaceManager(
             String functionNamespaceManagerName,
             String catalogName,
-            Map<String, String> properties)
+            Map<String, String> properties,
+            Optional<NodeManager> nodeManager)
```

Comment:
Why is it optional?

Reply:
Because `nodeManager` is only required while loading the `NativeFunctionNamespaceManager` and not the other function namespace managers, hence I thought making it optional is the right way.

Reply:
Can we supply it always and just have other namespace managers not use it?

Reply:
Also in some of the places where `loadFunctionNamespaceManager` is called I do not have access to an instance of the `NodeManager`.

Reply:
Resolving this conversation, changed `loadFunctionNamespaceManager` to take in NodeManager as a parameter.